### PR TITLE
RFC: Dev Docs

### PR DESCRIPTION
**Pull request by mltejera**
A proposal to update our dev docs.

Related PR scaffolding out new docs: #30483
Related Epic: #30642

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/c43a582b1e7587da4abbe03b04b1fc84f0f15415'>c43a582b1e7587da4abbe03b04b1fc84f0f15415</a></sub>
<div id="bundleSizeComment9423"/>

File: rfcs/react-components/components/dev-docs.md
Comment (brandonthomas) [inline rfcs/react-components/components/dev-docs.md:66]: Do we want to copy this over or just point to a deprecated link to the wiki? (If that's a thing GitHub supports). Mostly just curious if we have the time/resources to move all that content over.

File: rfcs/react-components/components/dev-docs.md
Comment (brandonthomas) [inline rfcs/react-components/components/dev-docs.md:40]: It would be good to go deeper here. To me, RFCs are good for gathering information/feedback or using as a proposal. I am not sure this is a good format for what we currently do so I'd like to see if everyone else would like to draw that line. Once something is approved and our current POR, move it out of RFC and into a durable document.

File: rfcs/react-components/components/dev-docs.md
Comment (brandonthomas) [inline rfcs/react-components/components/dev-docs.md:25]: What about process for documentation? Who should own documents for specific things? IMHO it's a good idea to update the documentation after you make a change that impacts them. Ex: we update our build or scaffolding process, that should include not just the changes but any documentation to explain the new normal.

File: rfcs/react-components/components/dev-docs.md
Comment (brandonthomas) [inline rfcs/react-components/components/dev-docs.md:14]: Do we want to throw any target metrics or times here? In the MADS repo we had the 10 minute goal from pulling the repo to being product even for new developers. Not saying we should use that goal but having that kind of a target for what we're doing might be a good idea.

File: rfcs/react-components/components/dev-docs.md
Comment (mltejera) [reply rfcs/react-components/components/dev-docs.md:66]: My intent was to just copy paste it over. 

I don't want to devote too many cycles to tidying it, but felt there might be some value in keeping it around.

But I do want to fully remove the wiki though as having it up will cause further confusion.

File: rfcs/react-components/components/dev-docs.md
Comment (mltejera) [reply rfcs/react-components/components/dev-docs.md:40]: nice - yeah, I like the idea of an 'accepted' RFC folder.

File: rfcs/react-components/components/dev-docs.md
Comment (mltejera) [reply rfcs/react-components/components/dev-docs.md:14]: Good callout. But I do want to be cautious about scope creep, so leaving out discussion of how to improve our dev infrastructure from this RFC.

Comment (brandonthomas) [reply rfcs/react-components/components/dev-docs.md:14]: Yah it was more about the spirit of it...so something like "as a new Fluent developer I can answer questions around making a new component in 2 minutes with existing docs" or something (don't copy that just threw it out there :D)

File: rfcs/react-components/components/dev-docs.md
Comment (AtishayMsft) [inline rfcs/react-components/components/dev-docs.md:20]: FYI, the charting team has started using docusaurus to host our technical documents. We coudn't find an easy way to publish our guides and dev docs in the existing fluent repo or the fluent docsite. 
https://microsoft.github.io/fluentui-charting-contrib/

Comment (mltejera) [reply rfcs/react-components/components/dev-docs.md:20]: Would a dedicated directory in the __docs__  help you all?

Comment (AtishayMsft) [reply rfcs/react-components/components/dev-docs.md:14]: More than the docs folder structuring we need a way for users to be able to easily navigate through the docs, search for specific terms, and interact/contribute with the docs content.

Comment (mltejera) [reply rfcs/react-components/components/dev-docs.md:14]: Awesome - these are great requirements. Thanks so much for the feedback and contributions Atishay!

File: rfcs/react-components/components/dev-docs.md
Comment (khmakoto) [inline rfcs/react-components/components/dev-docs.md:35]: I'd add one more:

```suggestion
- Allows for all dev document changes to go through proper review via PR review process.
```

File: rfcs/react-components/components/dev-docs.md
Comment (khmakoto) [inline rfcs/react-components/components/dev-docs.md:61]: Probably need to add one more for compat packages

```suggestion
- compat-packages.md // a doc explaining what a compat package is, how to create them, and the differences in maintenance when compared to stable packages
```

File: rfcs/react-components/components/dev-docs.md
Comment (khmakoto) [reply rfcs/react-components/components/dev-docs.md:67]: I think doing it the other way around would be helpful, copy over and then just leave one wiki article specifying that all dev-related documentation has moved into this folder.

**Review by khmakoto — APPROVED** [review]

File: rfcs/react-components/components/dev-docs.md
Comment (Hotell) [inline rfcs/react-components/components/dev-docs.md:29]: `__docs__` seems like something obscure, intended to be used by a machine, it would be good to go with human accessible naming, `documentation/`

**Review by Hotell — APPROVED** [review]
Tweaking the whole docs experience is a long time coming, unfortunately it was not on anyones priority list, so its great that someone can have time to improve it. thank you!

- +1 on stop using github wiki as it has its downsides
- I'd recommend to follow standard OSS practices in terms of having Contributing.md and Readme.md as a main crossroad documents ( which Github will suggest you while creating issues/PRs ) which would  then link to particular scoped docs (v0,v8,v9,wc).
- naming is hard, but I dont think this is the most human readable entry `__docs__/`

> GitHub integration:
>
> image


Important think to note is that not **all the docs can be public**, how would you propose handling those ?

File: rfcs/react-components/components/dev-docs.md
Comment (mltejera) [reply rfcs/react-components/components/dev-docs.md:29]: My intent with the underscores was that it keeps it at the very top of the file list :-)

Comment (mltejera) [reply rfcs/react-components/components/dev-docs.md:29]: maybe we can compromise and call it __documentation__ ?

Comment (ling1726) [inline rfcs/react-components/components/dev-docs.md:29]: Let's also follow the microsoft OSS pattern and use a `CONTRIBUTING.md` in the root of our repo. We can use relative links in the markdown file to whatever folder we choose for other documentation files

**Review by ling1726 — APPROVED** [review]

File: rfcs/react-components/components/dev-docs.md
Comment (mltejera) [reply rfcs/react-components/components/dev-docs.md:14]: Following up on this one:

Our goals for this effort are scope to contribution to the repo. We're going to stop short of trying to support outward or process related communication. Contributors are always encouraged to engage with PRs as they see fit.

File: rfcs/react-components/components/dev-docs.md
Comment (mltejera) [reply rfcs/react-components/components/dev-docs.md:27]: Good callout! We'll add some 'how to' docs along the way.

File: rfcs/react-components/components/dev-docs.md
Comment (mltejera) [reply rfcs/react-components/components/dev-docs.md:29]: Landed on `docs`

Comment (mltejera) [reply rfcs/react-components/components/dev-docs.md:29]: Yep, I like this a lot - we'll hold off on adding it until all the docs we want it to point to are in place.

**Review by smhigley — APPROVED** [review]

File: rfcs/react-components/components/dev-docs.md
Comment (AtishayMsft) [reply rfcs/react-components/components/dev-docs.md:63]: FYI, we have created a docusaurus based documentation site for the charting library. See here - https://microsoft.github.io/fluentui-charting-contrib/
This is inline with the practice followed by other open source projects in Mocrosoft and outside.

**Comment by mltejera** [comment]
Thanks for participating in this RFC folks - we're going to merge this and migrate things over.

**Pull request merged**
